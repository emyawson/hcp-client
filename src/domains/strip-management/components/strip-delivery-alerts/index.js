export * from './strip-delivery-alerts.component';
