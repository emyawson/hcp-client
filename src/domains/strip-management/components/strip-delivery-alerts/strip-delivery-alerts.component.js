import React from 'react';
import { isNil, all, pipe, not } from 'ramda';

import { colors } from 'src/core/styles/colors';
import {
  CheckmarkIcon,
  XIcon,
  FlagIcon,
  NoInformationIcon,
} from 'src/assets/icons';
import { translate } from 'src/i18n';

import {
  AlertContainer,
  AlertItem,
  AlertItemContent,
  AlertItemContentSpan,
} from './strip-delivery-alerts.style';

const checkmarkIcon = (
  <CheckmarkIcon
    fillColor={colors.trafficGreen}
    viewBoxWidth={45}
    viewBoxHeight={35}
    height={12}
  />
);

const notAvailableIcon = <NoInformationIcon height={9} width={9} />;
const xIcon = <XIcon height={9} width={9} />;

const flagIcon = <FlagIcon height={9} width={9} />; //eslint-disable-line no-unused-vars

const noAvailableValue = '-';
const passOrFailIcon = condition =>
  isNil(condition.pass)
    ? notAvailableIcon
    : condition.pass
      ? checkmarkIcon
      : xIcon;
const areValuesAvailable = all(
  pipe(
    isNil,
    not,
  ),
);

const createConditionAlert = (condition, valueDisplay, labelTextKey) => (
  <AlertItemContent>
    <AlertItemContentSpan>{passOrFailIcon(condition)}</AlertItemContentSpan>
    <AlertItemContentSpan>
      {`${translate(labelTextKey)} `}
      <strong>{valueDisplay}</strong>
    </AlertItemContentSpan>
  </AlertItemContent>
);

const createGlucoseThresholdAlertValueDisplay = condition =>
  areValuesAvailable([condition.threshold, condition.amount])
    ? `${condition.amount} ${translate('stripDelivery.alerts.outOf')} ${
        condition.threshold
      }`
    : noAvailableValue;

export const StripDeliveryAlert = ({
  conditions: { hypers, hypos, warnings, consumption },
  trafficLightStatus,
}) => (
  <AlertContainer>
    <AlertItem>
      {createConditionAlert(
        hypos,
        createGlucoseThresholdAlertValueDisplay(hypos),
        'stripDelivery.alerts.hypoThreshold',
      )}
      {createConditionAlert(
        hypers,
        createGlucoseThresholdAlertValueDisplay(hypers),
        'stripDelivery.alerts.upperLimit',
      )}
      {createConditionAlert(
        warnings,
        createGlucoseThresholdAlertValueDisplay(warnings),
        'stripDelivery.alerts.lowerLimit',
      )}
    </AlertItem>
  </AlertContainer>
);
