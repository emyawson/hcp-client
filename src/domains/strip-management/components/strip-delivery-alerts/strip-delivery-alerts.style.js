import styled from 'styled-components';

import { spacing, fontSize, colors } from 'src/core';
import { weight } from 'src/components/fonts/weights';

export const AlertHeader = styled.h3`
  margin: 0;
  padding: 0;
  padding-left: ${spacing.one};
  width: 100%;
  color: ${colors.black};
  font-size: ${fontSize.subheading};
  font-weight: ${weight.semiBold};
`;

export const AlertContainer = styled.div`
  padding: 0;
  display: flex;
  flex-wrap: wrap;
`;

export const AlertItem = styled.div`
  margin: ${spacing.two} 0 0;
`;

export const AlertItemBig = styled.div`
  margin: 0 0 ${spacing.three} 0;
  width: 100%;
`;

export const AlertItemContent = styled.p`
  padding-bottom: ${spacing.one};
  color: ${colors.black};
  font-size: ${fontSize.p};
  font-weight: ${weight.light};
`;

export const AlertItemContentSpan = styled.span`
  padding-right: ${spacing.two};
`;
