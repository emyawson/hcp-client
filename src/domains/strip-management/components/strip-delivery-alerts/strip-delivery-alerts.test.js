import React from 'react';
import { shallow } from 'enzyme';

import { StripDeliveryAlert } from './strip-delivery-alerts.component';

describe('Strip Delivery Alert ', () => {
  it('It renders correctly', () => {
    const defaultConditions = {
      hypers: { pass: false, threshold: null, amount: null },
      hypos: { pass: false, threshold: null, amount: null },
      warnings: { pass: false, threshold: null, amount: null },
      consumption: { pass: false, threshold: null, amount: null },
    };
    const wrapper = shallow(
      <StripDeliveryAlert conditions={defaultConditions} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
