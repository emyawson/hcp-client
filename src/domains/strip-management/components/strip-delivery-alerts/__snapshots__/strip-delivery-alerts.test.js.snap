// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Strip Delivery Alert  It renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    conditions={
      Object {
        "consumption": Object {
          "amount": null,
          "pass": false,
          "threshold": null,
        },
        "hypers": Object {
          "amount": null,
          "pass": false,
          "threshold": null,
        },
        "hypos": Object {
          "amount": null,
          "pass": false,
          "threshold": null,
        },
        "warnings": Object {
          "amount": null,
          "pass": false,
          "threshold": null,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.p>
          <styled.span>
            <Unknown
              height={9}
              width={9}
            />
          </styled.span>
          <styled.span>
            Hypo threshold alerts 
            <strong>
              -
            </strong>
          </styled.span>
        </styled.p>
        <styled.p>
          <styled.span>
            <Unknown
              height={9}
              width={9}
            />
          </styled.span>
          <styled.span>
            Upper limit alerts 
            <strong>
              -
            </strong>
          </styled.span>
        </styled.p>
        <styled.p>
          <styled.span>
            <Unknown
              height={9}
              width={9}
            />
          </styled.span>
          <styled.span>
            Lower limit alerts 
            <strong>
              -
            </strong>
          </styled.span>
        </styled.p>
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.p>
            <styled.span>
              <Unknown
                height={9}
                width={9}
              />
            </styled.span>
            <styled.span>
              Hypo threshold alerts 
              <strong>
                -
              </strong>
            </styled.span>
          </styled.p>,
          <styled.p>
            <styled.span>
              <Unknown
                height={9}
                width={9}
              />
            </styled.span>
            <styled.span>
              Upper limit alerts 
              <strong>
                -
              </strong>
            </styled.span>
          </styled.p>,
          <styled.p>
            <styled.span>
              <Unknown
                height={9}
                width={9}
              />
            </styled.span>
            <styled.span>
              Lower limit alerts 
              <strong>
                -
              </strong>
            </styled.span>
          </styled.p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.span>
                <Unknown
                  height={9}
                  width={9}
                />
              </styled.span>,
              <styled.span>
                Hypo threshold alerts 
                <strong>
                  -
                </strong>
              </styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown
                  height={9}
                  width={9}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "height": 9,
                  "width": 9,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Hypo threshold alerts ",
                  <strong>
                    -
                  </strong>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Hypo threshold alerts ",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "-",
                  },
                  "ref": null,
                  "rendered": "-",
                  "type": "strong",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.span>
                <Unknown
                  height={9}
                  width={9}
                />
              </styled.span>,
              <styled.span>
                Upper limit alerts 
                <strong>
                  -
                </strong>
              </styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown
                  height={9}
                  width={9}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "height": 9,
                  "width": 9,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Upper limit alerts ",
                  <strong>
                    -
                  </strong>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Upper limit alerts ",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "-",
                  },
                  "ref": null,
                  "rendered": "-",
                  "type": "strong",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.span>
                <Unknown
                  height={9}
                  width={9}
                />
              </styled.span>,
              <styled.span>
                Lower limit alerts 
                <strong>
                  -
                </strong>
              </styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <Unknown
                  height={9}
                  width={9}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "height": 9,
                  "width": 9,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Lower limit alerts ",
                  <strong>
                    -
                  </strong>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Lower limit alerts ",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "-",
                  },
                  "ref": null,
                  "rendered": "-",
                  "type": "strong",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.p>
            <styled.span>
              <Unknown
                height={9}
                width={9}
              />
            </styled.span>
            <styled.span>
              Hypo threshold alerts 
              <strong>
                -
              </strong>
            </styled.span>
          </styled.p>
          <styled.p>
            <styled.span>
              <Unknown
                height={9}
                width={9}
              />
            </styled.span>
            <styled.span>
              Upper limit alerts 
              <strong>
                -
              </strong>
            </styled.span>
          </styled.p>
          <styled.p>
            <styled.span>
              <Unknown
                height={9}
                width={9}
              />
            </styled.span>
            <styled.span>
              Lower limit alerts 
              <strong>
                -
              </strong>
            </styled.span>
          </styled.p>
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.p>
              <styled.span>
                <Unknown
                  height={9}
                  width={9}
                />
              </styled.span>
              <styled.span>
                Hypo threshold alerts 
                <strong>
                  -
                </strong>
              </styled.span>
            </styled.p>,
            <styled.p>
              <styled.span>
                <Unknown
                  height={9}
                  width={9}
                />
              </styled.span>
              <styled.span>
                Upper limit alerts 
                <strong>
                  -
                </strong>
              </styled.span>
            </styled.p>,
            <styled.p>
              <styled.span>
                <Unknown
                  height={9}
                  width={9}
                />
              </styled.span>
              <styled.span>
                Lower limit alerts 
                <strong>
                  -
                </strong>
              </styled.span>
            </styled.p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  <Unknown
                    height={9}
                    width={9}
                  />
                </styled.span>,
                <styled.span>
                  Hypo threshold alerts 
                  <strong>
                    -
                  </strong>
                </styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <Unknown
                    height={9}
                    width={9}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "height": 9,
                    "width": 9,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Hypo threshold alerts ",
                    <strong>
                      -
                    </strong>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Hypo threshold alerts ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "-",
                    },
                    "ref": null,
                    "rendered": "-",
                    "type": "strong",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  <Unknown
                    height={9}
                    width={9}
                  />
                </styled.span>,
                <styled.span>
                  Upper limit alerts 
                  <strong>
                    -
                  </strong>
                </styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <Unknown
                    height={9}
                    width={9}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "height": 9,
                    "width": 9,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Upper limit alerts ",
                    <strong>
                      -
                    </strong>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Upper limit alerts ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "-",
                    },
                    "ref": null,
                    "rendered": "-",
                    "type": "strong",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  <Unknown
                    height={9}
                    width={9}
                  />
                </styled.span>,
                <styled.span>
                  Lower limit alerts 
                  <strong>
                    -
                  </strong>
                </styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <Unknown
                    height={9}
                    width={9}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "height": 9,
                    "width": 9,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Lower limit alerts ",
                    <strong>
                      -
                    </strong>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Lower limit alerts ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "-",
                    },
                    "ref": null,
                    "rendered": "-",
                    "type": "strong",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
